domain/model/state/modelstate.go

```diff
@@ -112,25 +111,27 @@ func (s *ModelState) Delete(ctx context.Context, uuid coremodel.UUID) error {
 func (s *ModelState) Model(ctx context.Context) (coremodel.ReadOnlyModel, error) {
 	db, err := s.DB()
 	if err != nil {
-		return coremodel.ReadOnlyModel{}, errors.Trace(err)
+		return coremodel.ReadOnlyModel{}, errors.Capture(err)
 	}
 
 	m := dbReadOnlyModel{}
 	stmt, err := s.Prepare(`SELECT &dbReadOnlyModel.* FROM model`, m)
 	if err != nil {
-		return coremodel.ReadOnlyModel{}, errors.Annotatef(err, "preparing select read only model statement")
+		return coremodel.ReadOnlyModel{}, errors.Capture(err)
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		if err := tx.Query(ctx, stmt).Get(&m); errors.Is(err, sql.ErrNoRows) {
-			return fmt.Errorf("getting model read only information %w", modelerrors.NotFound)
-		} else if err != nil {
-			return fmt.Errorf("getting model read only information: %w", err)
+		err := tx.Query(ctx, stmt).Get(&m)
+		if errors.Is(err, sql.ErrNoRows) {
+			return errors.New("model does not exist").Add(modelerrors.NotFound)
 		}
-		return nil
+		return err
```

Comment:
Capture?

Reply:
It is traced below. We don't need to add more.